crates/transport-http/src/hyper_layer_transport.rs

```diff
@@ -0,0 +1,144 @@
+use alloy_json_rpc::{RequestPacket, ResponsePacket};
+use alloy_transport::{
+    utils::guess_local_url, TransportConnect, TransportError, TransportErrorKind, TransportFut,
+};
+use http_body_util::{BodyExt, Full};
+use hyper::{
+    body::{Buf, Bytes, Incoming},
+    header, Request, Response,
+};
+use hyper_util::client::legacy::Error;
+use std::{future::Future, marker::PhantomData, pin::Pin, task};
+use tower::Service;
+use tracing::{debug, debug_span, trace, Instrument};
+use url::Url;
+
+/// A [hyper] client that can be used with tower layers.
+#[derive(Clone, Debug)]
+pub struct HyperLayerTransport<S, B> {
+    url: Url,
+    service: S,
+    _pd: PhantomData<B>,
+}
+
+/// Alias for [`Request<Full<B>>`]
+pub type HyperRequest<B> = Request<Full<B>>;
+
+/// Alias for [`Response<Incoming>`]
+pub type HyperResponse = Response<Incoming>;
+
+/// Alias for pinned box future that results in [`HyperResponse`]
+pub type HyperResponseFut<T = HyperResponse, E = Error> =
+    Pin<Box<dyn Future<Output = Result<T, E>> + Send + 'static>>;
+
+impl<S, B> HyperLayerTransport<S, B>
```

Comment:
this now almost completely duplicates `HyperTransport` impl.

given that hyper's client implements `tower::Service` on it's own, I am wondering if we could unify this in `HyperTransport` by making underlying hyper client generic?